Framework/Algorithms/src/CarpenterSampleCorrection.cpp

```diff
@@ -0,0 +1,175 @@
+#include "MantidAlgorithms/CarpenterSampleCorrection.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/InstrumentValidator.h"
+#include "MantidAPI/Sample.h"
+#include "MantidAPI/SpectrumInfo.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/WorkspaceUnitValidator.h"
+#include "MantidDataObjects/EventWorkspace.h"
+#include "MantidGeometry/Instrument.h"
+#include "MantidGeometry/IComponent.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/Material.h"
+#include "MantidKernel/PhysicalConstants.h"
+
+#include <stdexcept>
+
+namespace Mantid {
+namespace Algorithms {
+DECLARE_ALGORITHM(CarpenterSampleCorrection) // Register the class
+                                             // into the algorithm
+                                             // factory
+
+using namespace Kernel;
+using namespace API;
+using Mantid::DataObjects::EventList;
+using Mantid::DataObjects::EventWorkspace;
+using Mantid::DataObjects::EventWorkspace_sptr;
+using Mantid::DataObjects::WeightedEventNoTime;
+using Mantid::HistogramData::HistogramX;
+using Mantid::HistogramData::HistogramY;
+using Mantid::HistogramData::HistogramE;
+using std::vector;
+using namespace Mantid::PhysicalConstants;
+using namespace Geometry;
+
+// Constants required internally only, so make them static. These are
+// Chebyshev expansion coefficients copied directly from Carpenter 1969 Table 1
+// clang-format off
+const std::string CarpenterSampleCorrection::name() const {
+  return "CarpenterSampleCorrection";
+}
+
+int CarpenterSampleCorrection::version() const { return 1; }
+
+const std::string CarpenterSampleCorrection::category() const {
+  return "CorrectionFunctions\\AbsorptionCorrections";
+}
+
+/**
+ * Initialize the properties to default values
+ */
+void CarpenterSampleCorrection::init() {
```

Comment:
Is this a subclass of the right thing that you could just use `copyProperty` ([example](https://github.com/mantidproject/mantid/blob/master/Framework/WorkflowAlgorithms/src/LoadEventAndCompress.cpp#L55))?

Reply:
No, this inherits from `DistributedAlgorithm` which does not have the `copyProperty` method. From the example, it looks to be a method that belongs to `DataProcessorAlgorithm`. 

I will change the parent class to `DistributedDataProcessorAlgorithm` to inherit from since it does have the `copyProperty` method.

Reply:
It looks like you may be able to inherit from [`DistributedDataProcessorAlgorithm`](https://github.com/mantidproject/mantid/blob/47b5ff5f004ad5198a9d749e1f39fb5f29aa5b7c/Framework/API/inc/MantidAPI/DataProcessorAlgorithm.h#L164) if that is more appropriate.